### ISSUE TITLE
Unconsistent magnifier checkbox state

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#20148]
The new "Enable magnifier" checkbox does not behave correctly upon NVDA Magnifier startup failure, e.g. when Windows Magnifier is already running.

There are various other issues with this checkbox and NVDA Magnifier failing to start. Please double check all these use cases.

### Steps to reproduce
* Start Windows Magnifier
* Try to start NVDA Magnifier; as expected, it should not start
* Open NVDA Magnifier settings
* Try to toggle "Enable Magnifier" checkbox
### Actual behavior
The checkbox "Enabe magnifier" is checked and cannot be unchecked
### Expected behavior
The checkbox "Enable Magnifier" should not be checked since NVDA Magnifier is not running.
### NVDA logs, crash dumps and other attachments
N/A
#### NVDA installed/portable/running from source
source
#### NVDA version
Last beta (commit 7bf0d9abd0acff5be5b6b328e83fd110978c4d7f)
#### Windows version
Windows 11 25H2 (AMD64) build 26200.8457
#### Name and version of other software in use when reproducing the issue
N/A
#### Other information about your system
N/A
#### Does the issue still occur after restarting your computer?
NOt tested
#### Have you tried any other versions of NVDA? If so, please report their behaviors
No. This is a new feature
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the System Accessibility Repair Tool in NVDA's tools menu?
Not tested , unrelated

**Comment by CyrilleB79** [comment]
Other issue with this checkbox:
* Start Windows Magnifier
* Open NVDA Magnifier settings
* Try to check the enable magnifier checkbox

Actual:
* The checkbox is checked
* NVDA says "Cannot start magnifier. Another magnifier application may already be running." immediately followed by "Starting Fullscreen magnifier with 10.0x zoom level, Normal filter, and Center full-screen mode"

Expected:
* The checkbox should not be checked.
* NVDA should not report "Starting Fullscreen magnifier" since it fails.
* Since this is a GUI action, the "Cannot start magnifier" should rather be in a message box, as done with screen curtain, rather than only spoken.

**Comment by CyrilleB79** [comment]
Cc @seanbudd 

I have not done so myself, but please also test:
* starting Windows Mag then NVDA with "Enable magnifier" already on in config
* Config containing both "Enable magnifier" enabled and screen curtain enabled. I do not know if this can be produced through the GUI but at least the config spec allows this.

**Comment by seanbudd** [comment]
Re-opening to track the extra changes suggested in https://github.com/nvaccess/nvda/pull/20184#issuecomment-4509072701 and https://github.com/nvaccess/nvda/issues/20148#issuecomment-4466624159

**Comment by Boumtchack** [comment]
can be closed as we merged https://github.com/nvaccess/nvda/pull/20503
